pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -1049,6 +1069,39 @@ func applyUpstreamTLSSettings(env *model.Environment, cluster *apiv2.Cluster, tl
 		}
 	}
 
+	// TODO where to store harcoded strings
+	if env.Mesh.GetEnableAutoMtls().Value {
+		// Per envoy docs
+		// If an endpoint metadata's value under *envoy.transport_socket* does not match any
+		// *TransportSocketMatch*, socket configuration fallbacks to use the *tls_context* or
+		// *transport_socket* specified in this cluster
+		cluster.TransportSocketMatches = []*apiv2.Cluster_TransportSocketMatch{
+			{
+				Name: "enableMtls",
```

Comment:
mtls

Reply:
should we consider to make this matching against our feature name? `enable_auto_mtls`? The match name here will eventually be showed up as envoy counter stats, be consistent might help customers to understand where the `auto` mode kicks in.